common/membership/ringpop.go

```diff
@@ -72,16 +81,45 @@ func (r *RingPop) Start(bootstrapHostPorts []string) {
 		return
 	}
 
-	bootParams := &swim.BootstrapOptions{
-		ParallelismFactor: 10,
-		JoinSize:          1,
-		MaxJoinDuration:   r.maxJoinDuration,
-		DiscoverProvider:  statichosts.New(bootstrapHostPorts...),
-	}
+	r.bootstrap(initialBootstrapHostPorts, bootstrapHostPostRetriever, bootstrapRetryBackoffInterval)
+}
+
+func (r *RingPop) bootstrap(
+	initialBootstrapHostPorts []string,
```

Comment:
Why initialBootstrapHostPorts is specifically passed in?
I think it makes the logic simpler if we just retrieve hostPorts from DB on each attempt including first attempt.

Reply:
great point. updated

Reply:
Awesome.  Update looks good.  Much cleaner.